python-avd/pyavd/_eos_designs/structured_config/underlay/router_bgp.py

```diff
@@ -19,155 +20,114 @@ class RouterBgpMixin(Protocol):
         if self.inputs.underlay_rfc5549 is True:
-            neighbor_interfaces = []
             for link in self._underlay_links:
                 if link["type"] != "underlay_p2p":
                     continue
 
-                neighbor_interfaces.append(
-                    {
-                        "name": link["interface"],
-                        "peer_group": self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-                        "remote_as": link["peer_bgp_as"],
-                        "peer": link["peer"],
-                        "description": "_".join([link["peer"], link["peer_interface"]]),
-                    },
+                self.structured_config.router_bgp.neighbor_interfaces.append_new(
+                    name=link["interface"],
+                    peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+                    remote_as=link["peer_bgp_as"],
+                    peer=link["peer"],
+                    description="_".join([link["peer"], link["peer_interface"]]),
                 )
 
                 if "subinterfaces" in link:
                     for subinterface in link["subinterfaces"]:
-                        vrfs_dict.setdefault(
-                            subinterface["vrf"],
-                            {
-                                "name": subinterface["vrf"],
-                                "router_id": self.shared_utils.router_id,
-                                "neighbor_interfaces": [],
-                            },
-                        )
-                        vrfs_dict[subinterface["vrf"]]["neighbor_interfaces"].append(
-                            {
-                                "name": subinterface["interface"],
-                                "peer_group": self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-                                "remote_as": link["peer_bgp_as"],
-                                # TODO: - implement some centralized way to generate these descriptions
-                                "description": f"{'_'.join([link['peer'], subinterface['peer_interface']])}_vrf_{subinterface['vrf']}",
-                            },
+                        # We need to add basic BGP VRF config in case the device is not covered by network_services. (Like a spine)
+                        if subinterface["vrf"] not in self.structured_config.router_bgp.vrfs:
+                            self.structured_config.router_bgp.vrfs.append_new(name=subinterface["vrf"], router_id=self.shared_utils.router_id)
+
+                        self.structured_config.router_bgp.vrfs[subinterface["vrf"]].neighbor_interfaces.append_new(
+                            name=subinterface["interface"],
+                            peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+                            remote_as=link["peer_bgp_as"],
+                            # TODO: - implement some centralized way to generate these descriptions
```

Comment:
Feat(eos_designs): Add centralized way to generate neighbor / neighbor interface descriptions